### ISSUE TITLE
me.Viewport.screenToWorld and worldToScreen are mislabeled?

### ISSUE THREAD

**Issue by parasyte** [melonjs/melonJS#271]
Mentioned on the forum: https://groups.google.com/forum/#!topic/melonjs/-LmnwhKt2Ik

The code for these functions does not look right, but because `screenToWorld` is actually used for converting pointer coordinates, then it must work for that specific case. I guess it's a useful system function, and may also be useful for overlaying HTML elements on the canvas screen. But I think the name for this functionality isn't quite right. It should mention "system" or "browser" or "agent", something to differentiate it from the viewport's sense of screen and world coordinates.

In other words, game developers want to convert between these coordinates:
![Screen vs world coordinates](http://1.bp.blogspot.com/-65DEv4Z_dMU/UVz5aHK4vwI/AAAAAAAAAJQ/vA4vzCh9QHM/s1600/coordinates_example.png)

And typically **NOT** these coordinates:
![Browser screen vs world coordinates](http://melonjs.github.io/docs/images/event_coord.png)
